### PR TITLE
go1.16: remove

### PR DESCRIPTION
**Pull request by ajs124**
###### Description of changes
go1.16 reached EOL with the release of go1.18

###### Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ajs124** [comment]
Result of `nixpkgs-review pr 168480` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>goofys</li>
    <li>gvisor</li>
    <li>utahfs</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>godu</li>
    <li>ipfs</li>
    <li>ipfs-migrator</li>
    <li>ipfs-migrator-all-fs-repo-migrations</li>
    <li>mycorrhiza</li>
    <li>ncdns</li>
    <li>obfs4</li>
    <li>onionshare</li>
    <li>onionshare-gui</li>
    <li>python310Packages.ipfshttpclient</li>
    <li>python310Packages.ipwhl</li>
    <li>python39Packages.ipfshttpclient</li>
    <li>python39Packages.ipwhl</li>
    <li>sampler</li>
    <li>textql</li>
    <li>vouch-proxy</li>
    <li>yajsv</li>
  </ul>
</details>

File: pkgs/applications/misc/sampler/default.nix
Comment (endocrimes) [inline pkgs/applications/misc/sampler/default.nix:14]: we should wait for a release or use an untagged commit - randomly pulling in patches sucks for everyone.

**Comment by zowoq** [comment]
@ofborg test ifps
@ofborg build godu ipfs mycorrhiza ncdns obfs4 onionshare onionshare-gui sampler textql vouch-proxy yajsv

File: pkgs/applications/virtualization/gvisor/default.nix
Comment (endocrimes) [inline pkgs/applications/virtualization/gvisor/default.nix:124]: latest builds with go1.17 - lets bump before landing this PR. https://github.com/google/gvisor/blob/master/go.mod#L3

**Review by endocrimes — CHANGES_REQUESTED** [review]
I see a lot of packages here that have go references removed without corresponding package bumps - have you verified that the software still works with newer go versions? (beyond just building?)

File: pkgs/development/tools/textql/default.nix
Comment (ajs124) [reply pkgs/development/tools/textql/default.nix:16]: done

File: pkgs/applications/virtualization/gvisor/default.nix
Comment (ajs124) [reply pkgs/applications/virtualization/gvisor/default.nix:124]: Can you bump it? I think I tried and it was non-trivial. IMO this is the maintainer's (@andrew-d) job not ours.

The release is ~10 months out of date, with a software that releases _very often_, looking at https://github.com/google/gvisor/tags.

File: pkgs/applications/misc/sampler/default.nix
Comment (ajs124) [reply pkgs/applications/misc/sampler/default.nix:14]: There isn't one, yet. We can wait, but then we'll need to keep this EOL go release around.

File: pkgs/development/tools/textql/default.nix
Comment (zowoq) [reply pkgs/development/tools/textql/default.nix]: > we should reference it via the fork 

Revert this change. If the fork is deleted the patch will disappear. Please add a comment linking the PR the commit came from.

Comment (ajs124) [reply pkgs/development/tools/textql/default.nix]: It'll be in the binary cache either way, so I don't really care, tbh.

The same diff can be gotten many ways, so how can we find a consensus here? I assume nixpkgs doesn't have a policy for this, despite there being dozens if not hundreds of occurrences of this throughout the codebase, as always?

Comment (zowoq) [reply pkgs/development/tools/textql/default.nix]: Using a source that won't disappear instead of one that may disappear seems link a better idea.

I've seen lots of reviews asking for a comment referencing the PR that a commit is sourced from, I haven't seen anyone ask that we actually use the PR authors fork to source the commits.

**Comment by Luflosi** [comment]
This breaks ipfs-migrator-all-fs-repo-migrations. See #161547, where I tested this and documented it in the PR description and commit message. Feel free to test this again as described in the commit message. Also see https://github.com/ipfs/fs-repo-migrations/issues/156, which I just created.

Please also run the IPFS test. Go 1.17 support seems to still be in progress: https://github.com/ipfs/go-ipfs/pull/8815.

**Comment by ajs124** [comment]
```
[…]
(finished: must succeed: ipfs --api /unix/run/ipfs.sock cat /ipfs/QmfNnecR7tB17fa4tjY9V2dDVwsanVnJPws3HoEM9W2wjC | grep fnord2, in 0.17 seconds)
machine: must succeed: test -e /mnt/ipfs/config
(finished: must succeed: test -e /mnt/ipfs/config, in 0.02 seconds)
machine: must succeed: test ! -e /var/lib/ipfs/
(finished: must succeed: test ! -e /var/lib/ipfs/, in 0.02 seconds)
(finished: run the VM test script, in 42.02 seconds)
test script finished in 42.09s
cleanup
kill machine (pid 8)
machine # qemu-kvm: terminating on signal 15 from pid 6 (/nix/store/bvzpppawf2naqv3qhxqv66jxaq3iaxyj-python3-3.9.11/bin/python3.9)
(finished: cleanup, in 0.02 seconds)
kill vlan (pid 7)
/nix/store/ciff928nb15cnj17jhlhbk5klnpgb08r-vm-test-run-ipfs
```
so `ipfs` is fine, but `ipfs-migrator-all-fs-repo-migrations` is apparently still broken.

**Comment by Luflosi** [comment]
See https://github.com/ipfs/go-ipfs/pull/8815 for why the currently existing migrations probably won't be updated to support a newer Go version. This means, we either need to keep Go 1.16 or remove the incompatible migrations. In the second case I could write a stub to replace them, which prints some text, explaining the situation.

**Comment by ajs124** [comment]
> In the second case I could write a stub to replace them, which prints some text, explaining the situation.

@Luflosi can you go ahead and do that? go1.16 is EOL and I'd very much like to see it removed before branch-off.

**Comment by dasJ** [comment]
I'd like to remind the people here that branch-off is scheduled for tomorrow and depending on how good the staging builds progress, we will branch off rather soon.

File: pkgs/top-level/all-packages.nix
Comment (dasJ) [inline pkgs/top-level/all-packages.nix:8633]: Does this not need an alias?

Comment (mweinelt) [reply pkgs/top-level/all-packages.nix:8633]: A throw to be precise.

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: Done.

File: pkgs/tools/filesystems/goofys/default.nix
Comment (mausch) [inline pkgs/tools/filesystems/goofys/default.nix:34]: ping @adisbladis , I don't think you've been notified?

Comment (adisbladis) [reply pkgs/tools/filesystems/goofys/default.nix:34]: Thanks for the ping. Fixed in https://github.com/NixOS/nixpkgs/pull/176255.

**Pull request merged**
